src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1051,20 +1051,62 @@ object Desugar {
 
       // encode arguments
       val dArgs = {
-        val parameterCount = p.callee match {
+        val params: Vector[Type] = p.callee match {
           // `BuiltInFunctionKind` has to be checked first since it implements `Symbolic` as well
-          case f: ap.BuiltInFunctionKind => getBuiltInFuncType(f).args.length
+          case f: ap.BuiltInFunctionKind => getBuiltInFuncType(f).args
           case base: ap.Symbolic => base.symb match {
-            case fsym: st.WithArguments => fsym.args.length
+            case fsym: st.WithArguments => fsym.args.map(fsym.context.typ(_))
             case c => Violation.violation(s"This case should be unreachable, but got $c")
           }
         }
-        sequence(p.args map exprD(ctx)).map {
+
+        val parameterCount: Int = params.length
+
+        // is of the form Some(x) if the type of the last param is variadic and the type of its elements is x
+        val variadicTypeOption: Option[Type] = params.lastOption match {
+          case Some(VariadicT(elem)) => Some(elem)
+          case _ => None
+        }
+
+        val wRes: Writer[Vector[in.Expr]] = sequence(p.args map exprD(ctx)).map {
           // go function chaining feature
           case Vector(in.Tuple(targs)) if parameterCount > 1 => targs
           case dargs => dargs
         }
+
+        variadicTypeOption match {
+          case Some(variadicTyp) => for {
+            res <- wRes
+            variadicInTyp = typeD(variadicTyp, Addressability.sliceElement)(src)
+            len = res.length
+            argList = res.lastOption.map(_.typ) match {
+              case Some(in.SliceT(elems, _)) if len == parameterCount && elems == variadicInTyp =>
+                // corresponds to the case where an unpacked slice is already passed as an argument
+                res
+              case Some(in.TupleT(_, _)) if len == 1 && parameterCount == 1 =>
+                // supports chaining function calls with variadic functions of one argument
+                val argsMap: Map[BigInt, in.Expr] =
+                  res.last.asInstanceOf[in.Tuple].args.zipWithIndex.map {
+                    case (arg, index) => BigInt(index) -> arg
+                  }.toMap
+                Vector(in.SliceLit(variadicInTyp, argsMap)(src))
+              case _ if len >= parameterCount =>
```

Comment:
I think just using `len >= parameterCount - 1` deals with both cases.

Reply:
I would still need to do a case split to generate `nil` when `len == parameterCount - 1`. I don't think that I gain anything from merging the two

Reply:
My comment was for the case that you can join both cases, which you explained to not be the case.